software/controller/test/frame_detector/frame_detector_test.cpp

```diff
@@ -0,0 +1,283 @@
+/* Copyright 2020-2021, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "frame_detector.h"
+
+#include <cmath>
+#include <string>
+#include <vector>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "proto_traits.h"
+#include "soft_rx_buffer.h"
+
+using namespace std;
+
+constexpr uint32_t LEN = 20;
+
+using State = FrameDetector<SoftRxBuffer<LEN>, LEN>::State;
+using FrameDetectorTest = FrameDetector<SoftRxBuffer<LEN>, LEN>;
```

Comment:
All these tests use the same length for the rx buffer and the frame detector's length. Is that a property we expect in the controller? If so, could we assert that somehow in the template?

Reply:
I do not think that is a given. I think in practice we might have a rather large buffer for lots of quiet time but the frame buffer would be only as large as the largest expected message?